### PR TITLE
Fix Query Actions in Ballerina Shell

### LINKED ISSUE #34824: Query Action is not working in Ballerina Shell
**Description:**
Try the following code in Ballerina shell.

> ```
> type Student record {
>     string firstName;
>     string lastName;
>     float score;
> };
> ```
> 
> ```
> type FullName record {|
>     string firstName;
>     string lastName;
> |};
> ```
> 
> `Student s1 = {firstName: "Alex", lastName: "George", score: 1.5};`
> `Student s2 = {firstName: "John", lastName: "David", score: 0.9};`
> `Student s3 = {firstName: "Jane", lastName: "Smith", score: 2};`
> 
> `Student[] studentList = [s1, s2, s3];`
> `FullName[] nameList = [];`
> `error? e = from var student in studentList do { nameList.push( {firstName: student.firstName, lastName: student.lastName}); };`

However, if this statement is embedded within a function in the shell as follows, it would be executed without such errors. 
```
function fn() returns error? {
    error? e = from var student in studentList do { nameList.push({firstName: student.firstName, lastName: student.lastName}); };
    return  e;
}
```

**Steps to reproduce:**
image

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
Fixes #34824

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java [hunk 1]
```diff
@@ -136,18 +145,46 @@ public VariableDeclarationSnippet createVariableDeclarationSnippet(Node node) {
...
             // It is the only possible qualifier that can be transferred.
             if (varNode.finalKeyword().isPresent()) {
                 qualifiers = NodeFactory.createNodeList(varNode.finalKeyword().get());
             }
+
+            if (((VariableDeclarationNode) node).initializer().get().kind() == SyntaxKind.QUERY_ACTION) {
+                TypedBindingPatternNode typedBindingPatternNode = varNode.typedBindingPattern();
+                TypeDescriptorNode typeDescriptorNode = typedBindingPatternNode.typeDescriptor();
+                BindingPatternNode bindingPatternNode = typedBindingPatternNode.bindingPattern();
+
```
Comment (sanjana) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java:160]:
```suggestion
```

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java [hunk 2]
```diff
@@ -136,18 +145,46 @@ public VariableDeclarationSnippet createVariableDeclarationSnippet(Node node) {
...
+                String functionPart = "function() returns ";
+                String functionBody = varNode.initializer().get().toString();
+
```
Comment (sanjana) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java:163]:
```suggestion
```

**Review by sanjana — CHANGES_REQUESTED** [review]
Minor formatting suggestions.

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java [hunk 3]
```diff
@@ -40,43 +41,60 @@
  */
 public abstract class SnippetFactory extends DiagnosticReporter {
     /**
-     * Creates a snippet from the given node.
+     * Creates a snippets from the given node.
```
Comment (sanjana) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java:44]:
```suggestion
     * Creates snippets from the given node.
```

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java [hunk 4]
```diff
@@ -40,43 +41,60 @@
...
         for (Node node : nodes) {
-            snippets.add(createSnippet(node));
+            snippets.addAll(createSnippets(node));
         }
         return snippets;
     }
 
     /**
-     * Creates a snippet from the given node.
+     * Creates a snippets from the given node.
```
Comment (sanjana) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java:60]:
```suggestion
     * Creates snippets from the given node.
```

Ditto

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java [hunk 5]
```diff
@@ -94,13 +112,13 @@ public Snippet createSnippet(Node node) throws SnippetException {
     public abstract ImportDeclarationSnippet createImportSnippet(Node node);
 
     /**
-     * Create a variable declaration snippet from the given node.
+     * Create a variable declaration snippets from the given node.
```
Comment (sanjana) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/SnippetFactory.java:115]:
```suggestion
     * Create variable declaration snippets from the given node.
```

Ditto.

**Review by sanjana — DISMISSED** [review]
Simple grammatical errors.

**Review by hasithaa — APPROVED** [review]

**Pull request merged**